test/_functions/common/event-recorder/java/Handler.java

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import io.nuclio.Context;
+import io.nuclio.Event;
+import io.nuclio.EventHandler;
+import io.nuclio.Response;
+
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Handler implements EventHandler {
+
+    private static final String EVENTS_LOG_FILE_PATH = "/tmp/events.json";
+
+    @Override
+    public Response handleEvent(Context context, Event event) {
+        try {
+            String triggerKind = ensureString(event.getTriggerInfo().getKindName());
+            if (!"http".equals(triggerKind) || invokedByCron(event)) {
+                String body = ensureString(event.getBody());
+                context.getLogger().debug("Received event with body: " + body);
+
+                // Serialize record
+                Map<String, Object> record = new HashMap<>();
+                record.put("body", body);
+                record.put("headers", ensureHeaders(event.getHeaders()));
+                record.put("timestamp", Instant.now().toString());
+
+                String serializedRecord = new JSONObject(record).toString();
+
+                // Append record to log file
+                appendToLogFile(serializedRecord);
+                return new Response();
+            }
+
+            // Read log file and return events
+            String encodedEventLog = getEventsLog();
+            context.getLogger().debug("Returning events: " + encodedEventLog);
+            return new Response().setBody(encodedEventLog).setStatusCode(200);
+        } catch (Exception e) {
+            context.getLogger().error("Error handling event", e);
+            return new Response().setBody(e.toString()).setStatusCode(500);
+        }
+    }
+
+    private boolean invokedByCron(Event event) {
+        String header = getHeader(event, "x-nuclio-invoke-trigger");
+        return "cron".equals(header);
```

Comment:
is this case insensitive?

Reply:
@TomerShor generally, headers are, so I think here i should add some lower() to ensure that cases are aligned